### ISSUE TITLE
DLNA on older Samsung TV shows empty video folder

### ISSUE THREAD

**Issue by MartenBE** [jellyfin/jellyfin#2335]
Hi,

I am trying to watch shows on a Samsung TV using DLNA. The TV finds my jellyfin server without problem, and I checked that port 1900/udp is (I) only used by jellyfin and (II) allowed through my firewall.

Unfortunately, when accessing the jellyfin instance on the Samsung TV there seems to be no content.

Steps taken:

* Select the jellyfin instance found on the Samsung TV: https://imgbox.com/0KgTzUIf
* The TV displays 3 choices: Videos, Images and Audio. These are not folders on my jellyfin server (default choices by the TV for any USB/server/...): https://imgbox.com/olKeILB3
* After selecting one of these folders, the TV displays an error that no content is found: https://imgbox.com/4KE0kx5Z

The logs show the following when selecting the videos choice:

    [2020-01-24 18:29:35.373 +01:00] [ERR] Error parsing item Id: "V". Returning user root folder.

When selecting the image or audio folder, the same error message appears, but the displayed letter is different:

    [2020-01-24 18:32:25.049 +01:00] [ERR] Error parsing item Id: "I". Returning user root folder.
    [2020-01-24 18:30:05.493 +01:00] [ERR] Error parsing item Id: "A". Returning user root folder.

How can I fix this?

**Comment by PrplHaz4** [comment]
There is a good chance this will be fixed in the next release via https://github.com/jellyfin/jellyfin/issues/2164

**Comment by MartenBE** [comment]
Wel ... the jellyfin instance now detects my folders, but the folders show up empty

**Comment by yriveiro** [comment]
Same here,

This is what the log shows:

`[2020-03-15 15:47:24.519 +00:00] [ERR] Error parsing item Id: "V". Returning user root folder.`

**Comment by PrplHaz4** [comment]
DLNA debug logging works now - any chance you can try turning it on to see what you're getting for a BROWSE or BROWSEDIRECTCHILDREN response?

**Comment by MartenBE** [comment]
> DLNA debug logging works now - any chance you can try turning it on to see what you're getting for a BROWSE or BROWSEDIRECTCHILDREN response?

Could you show me where I can see this? It doesn't show up in the log.
@PrplHaz4 Something curious: when experimenting with a test folder with a movie in a subfolder, I noticed that I could get DLNA to play the movie if the library is set on "Mixed Content". If I recreate the library on the same folder with type "Movies" or "Series", the folder shows up empty.

**Comment by PrplHaz4** [comment]
@MartenBE 
1. Enable Debug Logging: https://jellyfin.org/docs/general/administration/troubleshooting.html#debug-logging
2. Jellyfin Admin Dashboard -> DLNA -> Enable DLNA Debug Logging

Very strange about mixed content - do you happen to have something in that has a `:` (colon) in the title on a Samba share (or something where the filename will be hard to parse)?

**Comment by MartenBE** [comment]
The logging adds nothing new:

```
jellyfin_1  | [09:27:59] [INF] Jellyfin version: 10.5.3
jellyfin_1  | [09:27:59] [INF] Arguments: ["/jellyfin/jellyfin.dll", "--datadir", "/config", "--cachedir", "/cache", "--ffmpeg", "/usr/local/bin/ffmpeg"]
jellyfin_1  | [09:27:59] [INF] Operating system: Linux
jellyfin_1  | [09:27:59] [INF] Architecture: X64
jellyfin_1  | [09:27:59] [INF] 64-Bit Process: True
jellyfin_1  | [09:27:59] [INF] User Interactive: True
jellyfin_1  | [09:27:59] [INF] Processor count: 8
jellyfin_1  | [09:27:59] [INF] Program data path: /config
jellyfin_1  | [09:27:59] [INF] Web resources path: /jellyfin/jellyfin-web
jellyfin_1  | [09:27:59] [INF] Application directory: /jellyfin/
jellyfin_1  | [09:28:00] [INF] Setting cache path: /cache
jellyfin_1  | [09:28:00] [INF] Loading assemblies
jellyfin_1  | [09:28:00] [INF] Loaded assembly Jellyfin.Plugin.OpenSubtitles, Version=7.0.0.0, Culture=neutral, PublicKeyToken=null from /config/plugins/Open Subtitles/Jellyfin.Plugin.OpenSubtitles.dll
jellyfin_1  | [09:28:00] [INF] Loaded assembly OpenSubtitlesHandler, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null from /config/plugins/Open Subtitles/OpenSubtitlesHandler.dll
jellyfin_1  | [09:28:02] [INF] Kestrel listening on all interfaces
jellyfin_1  | [09:28:02] [INF] Running startup tasks
jellyfin_1  | [09:28:02] [INF] Daily trigger for Extract Chapter Images set to fire at 04/10/2020 02:00, which is 16:31:57.0523269 from now.
jellyfin_1  | [09:28:02] [INF] Found ffmpeg version 4.2.1
jellyfin_1  | [09:28:03] [INF] Available decoders: ["h264", "h264_cuvid", "hevc", "hevc_cuvid", "mpeg2video", "aac", "ac3", "mp3"]
jellyfin_1  | [09:28:03] [INF] Available encoders: ["libx264", "h264_nvenc", "h264_v4l2m2m", "h264_vaapi", "libx265", "hevc_nvenc", "hevc_vaapi", "mpeg4", "msmpeg4", "libvpx", "libvpx-vp9", "aac", "ac3", "libmp3lame", "libopus", "libvorbis", "srt"]
jellyfin_1  | [09:28:03] [INF] FFmpeg: SetByArgument: /usr/local/bin/ffmpeg
jellyfin_1  | [09:28:03] [INF] ServerId: 492e7bc32392481fb84c907c60b7c03d
jellyfin_1  | [09:28:03] [INF] Executed all pre-startup entry points in 0:00:00.2380686
jellyfin_1  | [09:28:03] [INF] Core startup complete
jellyfin_1  | [09:28:03] [INF] Registering publisher for urn:schemas-upnp-org:device:MediaServer:1 on 192.168.1.211
jellyfin_1  | [09:28:03] [INF] Watching directory /new
jellyfin_1  | [09:28:03] [INF] Watching directory /series
jellyfin_1  | [09:28:03] [INF] Executed all post-startup entry points in 0:00:00.3180833
jellyfin_1  | [09:28:03] [INF] Startup complete 0:00:04.0985496
jellyfin_1  | [09:28:03] [INF] Watching directory /movies
jellyfin_1  | [09:28:05] [INF] StartupTrigger fired for task: Update Plugins
jellyfin_1  | [09:28:05] [INF] Queueing task PluginUpdateTask
jellyfin_1  | [09:28:05] [INF] Executing Update Plugins
jellyfin_1  | [09:28:06] [INF] Update Plugins Completed after 0 minute(s) and 0 seconds
jellyfin_1  | [09:28:06] [INF] ExecuteQueuedTasks
jellyfin_1  | [09:28:12] [ERR] Error parsing item Id: V. Returning user root folder.
jellyfin_1  | [09:28:12] [ERR] Error parsing item Id: V. Returning user root folder.
```

There are no colons in any filename. However, when I replace the library by a mixed type library, it (almost) finds them all! I think the bug should be located in this.

**Comment by irovder** [comment]
Same problem on a Sony TV (2017 non-android). Video folders are there but no videos. Jellyfin 10.5.5 arm (Odroid C1). Other DLNA sources work OK on that TV.

**Comment by LinFor** [comment]
I have exactly the same issue as author. I did some investigations and seems I found root cause of problem.

TV goes into error state and stop browsing when receives dlna response with `NumberReturned ` > `RequestedCount`. This happens due to lack of paging code when returning "virtual" folders inside root media folders: https://github.com/jellyfin/jellyfin/blob/06caee28b7f81fb6898a771df5a5064fdf94891d/Emby.Dlna/ContentDirectory/ControlHandler.cs#L757-L815
Same effect in other media folders (TV, Music) and some virtual folders (passed query.Limit ignored and overwrited): https://github.com/jellyfin/jellyfin/blob/06caee28b7f81fb6898a771df5a5064fdf94891d/Emby.Dlna/ContentDirectory/ControlHandler.cs#L1047-L1052
(This can be also the cause of #2688)

"`Error parsing item Id: "V"`" shown in dlna logs doesn't interfere with this issue, it goes from https://github.com/jellyfin/jellyfin/blob/06caee28b7f81fb6898a771df5a5064fdf94891d/Emby.Dlna/ContentDirectory/ControlHandler.cs#L298-L300
And these static IDs ignored when requestID parsing:
https://github.com/jellyfin/jellyfin/blob/06caee28b7f81fb6898a771df5a5064fdf94891d/Emby.Dlna/Didl/DidlBuilder.cs#L630-L634
This also can be fixed, but doesn't reflects with TV behavior.

I corrected this behavior and made successful tests on real TV, but currently can't make PR.

PS: @BaronGreenback, @Bond-009, @cvium can you reopen this issue and make a fix?

**Comment by BaronGreenback** [comment]
@LinFor: Am currently debugging a dlna server PR on my local machine.

Have already discovered and corrected a limit issue when "special folders" are not counted in the results, and also the limit being overwritten.

Have to admit i'm not familar with the feature list - so i'll go and take a look at the specs on that one.

**Comment by BaronGreenback** [comment]
Haven't got a samsung - but according to what i've just read - it lookings like the id's should be numeric.


```
samsung.com_BASICVIEW feature lists four container types
<container id="1" type="object.item.imageItem"/>
<container id="1" type="object.item.audioItem"/>
<container id="1" type="object.item.videoItem"/>
<container id="1" type="object.item.textItem"/>
```

**Comment by LinFor** [comment]
> Haven't got a samsung - but according to what i've just read - it lookings like the id's should be numeric.
> @LinFor any chance you could make this change and see what effect it has?
TV just passes received `container id` got from features request to `browse` handler. So currently all works correctly except error message in log - just because jellyfin doesn't expect "V", "A", "I" ids in browse handler.

I'll test changing these ids to zeroes (I think 0 will be better than 1, as default IsIdRoot logic expects zero id as root), but I didn't expect any issues with it.

PS: Will be great if DLNA code can use ids provided by features to select only media folders containing this type of media-resources.
Currently in TV I selects "Video" (in native TV menu, got from features), than it passes "V" identifier to jellyfin, next TV shows me root jellyfin selection (Movies, TV, Music and Live in my case) - it looks illogical (I made the same selection twice) and shows deliberately incorrect paths in menu (If I under "Video" selection browse to Music folders - I see only folders without any playable content in it - due to container type mismatch).

**Comment by LinFor** [comment]
> @LinFor any chance you could make this change and see what effect it has?

Made tests on real TV (Samsung *UE6* from 2012 product line).
Features with zero ids and added textItem works well. My code block now:
```
            builder.Append("<container id=\"0\" type=\"object.item.imageItem\"/>");
            builder.Append("<container id=\"0\" type=\"object.item.audioItem\"/>");
            builder.Append("<container id=\"0\" type=\"object.item.videoItem\"/>");
            builder.Append("<container id=\"0\" type=\"object.item.textItem\"/>");
```

Also some suggestions to your PR:
1. Make up for the lack of logging SOAP headers here (HTTP transport headers logged in ProcessControlRequestAsync, but inner SOAP headers including soap action and parameters not logged at all): https://github.com/jellyfin/jellyfin/blob/06caee28b7f81fb6898a771df5a5064fdf94891d/Emby.Dlna/Service/BaseControlHandler.cs#L67
2. My TV does not recognize content dates in full ISO 8601 format, but recognizes short dates with "yyyy-MM-dd" format: https://github.com/jellyfin/jellyfin/blob/06caee28b7f81fb6898a771df5a5064fdf94891d/Emby.Dlna/Didl/DidlBuilder.cs#L734
Format docs shows that yyyy-MM-dd allowed ([#1, p.152, dc:date](https://openconnectivity.org/wp-content/uploads/2015/11/UPnP-av-ContentDirectory-v4-Service-20150319.pdf); [#2](https://www.dublincore.org/specifications/dublin-core/dcmi-terms/#http://purl.org/dc/elements/1.1/date)), so reducing the accuracy to a day seems to me quite appropriate for the sake of compatibility.

**Comment by mr-simba** [comment]
Hello,

I also have a non working dlna on a samsung tv (only emtpy folders). 

Can I change the above mentioned files in a docker container? Where can I find these files?
When will the changes be in the official jellyfin docker?

**Comment by quile-v** [comment]
HI,

I have same problems as above. On my older samsung smart TV I am only getting empty folders. 

Log also shows:
[ERR] Error parsing item Id: V. Returning user root folder.

I am running jellyfin on windows 10 and cannot find the file ControlHandler.cs and folders as mentioned above.
What file do I need to change to try in the windows 10 instalation of Jellyfin to get it working on my older samsung TV?

Thank you.

**Comment by cvium** [comment]
Should be fixed by #6795
